src/ServiceClient/Implements/SocketClient/DeribitSocketClient.cs

```diff
@@ -103,21 +114,41 @@ public async Task CheckAvailabilityAsync(CancellationToken token)
         await ReadAsync<object>(token);
     }
 
-    public async Task AuthenticateAsync(CancellationToken token)
+    public Task AuthenticateAsync(CancellationToken token)
     {
-        if (webSocket.State != WebSocketState.Open)
-        {
-            await ConnectAsync(token);
-        }
         var data = new
         {
             grant_type = "client_credentials",
             client_id = deribitOptions.ClientId,
             client_secret = deribitOptions.ClientSecret,
         };
+        return AuthenticateWithDataAsync(data, token);
+    }
+
+    private Task AuthenticateWithRefreshTokenAsync(CancellationToken token)
+    {
+        var data = new
+        {
+            grant_type = "refresh_token",
+            refresh_token = Credentials?.RefreshToken ?? string.Empty,
```

Comment:
I don't think we can be authenticated using an empty token

Why the _RefreshToken_ will be empty (due to an error trying to get the token)? In that case we need to try again to get the token or abort the app execution with a graceful message. Anyways the Deribit API from now on will always respond with an error (we're not authenticated anymore)